developer/system/template-manager.md

```diff
@@ -0,0 +1,259 @@
+---
+layout: doc
+title: Template Manager
+permalink: /doc/template-manager/
+---
+
+Template Manager Design
+=======================
+
+Motivation
+----------
+
+Previously, template VMs were distributed by RPM packages and managed by `yum`/`dnf`.
+However, tracking inherently dynamic VM images with a package manager suited for static files creates some challenges.
+For example, users may accidentally update the images, overriding local changes ([#996], [#1647]).
+(Or in the case of [#2061], want to specifically override the changes.)
+Other operations that work well on normal VMs are also somewhat inconsistent on RPM-managed templates.
+This includes actions such as renaming ([#839]), removal ([#5509]) and backup/restore ([#1385], [#1453], [discussion thread 1], [discussion thread 2]).
+In turn, this creates inconveniences and confusion for users ([#1403], [#4518]).
+
+Also, the usage of RPM packages meant that installing a template results in arbitrary code execution, which is not ideal.
+
+Besides distribution, users may also wish to have an integrated template management application ([#2062], [#2064], [#2534], [#3040]), as opposed to the situation where multiple programs are required for different purposes, e.g., `qubes-dom0-update`, `dnf`, `qvm-remove`, `qubes-manager`.
+
+To tackle these issues, `qvm-template` is created.
+It strives to provide not only a better mechanism for handling template installation but also a consistent user-facing interface to deal with template management.
+
+[#996]: https://github.com/QubesOS/qubes-issues/issues/996
+
+[#1647]: https://github.com/QubesOS/qubes-issues/issues/1647
+
+[#2061]: https://github.com/QubesOS/qubes-issues/issues/2061
+
+[#839]: https://github.com/QubesOS/qubes-issues/issues/839
+
+[#5509]: https://github.com/QubesOS/qubes-issues/issues/5509
+
+[#1385]: https://github.com/QubesOS/qubes-issues/issues/1385
+
+[#1453]: https://github.com/QubesOS/qubes-issues/issues/1453
+
+[#1403]: https://github.com/QubesOS/qubes-issues/issues/1403
+
+[#4518]: https://github.com/QubesOS/qubes-issues/issues/4518
+
+[#2062]: https://github.com/QubesOS/qubes-issues/issues/2062
+
+[#2064]: https://github.com/QubesOS/qubes-issues/issues/2064
+
+[#2534]: https://github.com/QubesOS/qubes-issues/issues/2534
+
+[#3040]: https://github.com/QubesOS/qubes-issues/issues/3040
+
+[discussion thread 1]: https://groups.google.com/forum/#!topic/qubes-devel/rwc2_miCNNE/discussion
+
+[discussion thread 2]: https://groups.google.com/forum/#!topic/qubes-users/uQEUpv4THsY/discussion
+
```

Comment:
The extra blank lines between these are not necessary. They don't mess anything up, but they take up a lot of space.